runconfig/parse_unix.go

```diff
@@ -7,51 +7,53 @@ import (
 	"strings"
 )
 
-func parseNetMode(netMode string) (NetworkMode, error) {
-	parts := strings.Split(netMode, ":")
+// ValidateNetMode ensures that the various combinations of requested
+// network settings are valid.
+func ValidateNetMode(c *Config, hc *HostConfig) error {
+	// We may not be passed a host config, such as in the case of docker commit
+	if hc == nil {
+		return nil
+	}
```

Comment:
@jhowardmsft "default" was introduced exactly to make sure we dont break older versions when we move towards the new driver functionality and have a ability to provide a pluggable default at the daemon level.
Essentially, we didnt have to change anything on the Application to introduce a new feature in the daemon.

am afraid the current changes in this PR breaks that promise (whether it is "default" or "bridge"). 
I will let @icecrime @tiborvass @cpuguy83 & others to comment on this particular point.